inst/scribble/xaringanExtra-scribble.js

```diff
@@ -0,0 +1,490 @@
+  }
+
+  getVisibleSlideCanvas() {
+    return this.fabrics[slideshow.getCurrentSlideIndex()]
+  }
+
+  getVisibleSlideOuterContainer() {
+    const outerDiv = this.getVisibleSlide().querySelector(
+      '.xe-scribble.outer-canvas-container',
+    )
+    return outerDiv
+  }
+
+  getVisibleSlideCanvasContainers() {
+    const canvasDiv = this.getVisibleSlide().querySelectorAll(
+      '.xe-scribble.drawing-canvas',
+    )
+    return canvasDiv
+  }
+
+  createCanvas(id) {
+    const slideSize = this.getVisibleSlideSize()
+
+    const canvasDiv = document.createElement('div')
+    canvasDiv.classList.add('xe-scribble', 'outer-canvas-container')
+    canvasDiv.style.width = slideSize.width
+    canvasDiv.style.height = slideSize.height
+    canvasDiv.style.left = slideSize.left + 'px'
+    canvasDiv.style.top = slideSize.top + 'px'
+    ;['mousedown', 'touchstart'].forEach((gesture) => {
+      canvasDiv.addEventListener(gesture, this.mousedown.bind(this))
+    })
+    ;['mouseup', 'touchend'].forEach((gesture) => {
+      canvasDiv.addEventListener(gesture, this.mouseup.bind(this))
+    })
+
+    const canvas = document.createElement('canvas')
+    canvas.setAttribute('id', 'canvas' + id)
+    canvas.classList.add('xe-scribble', 'drawing-canvas')
+    canvas.setAttribute('width', slideSize.width)
+    canvas.setAttribute('height', slideSize.height)
+
+    canvasDiv.appendChild(canvas)
+
+    return canvasDiv
+  }
+
+  addCanvasToAllSlides() {
+    // Add <canvas> to slide container
+    const nslides = slideshow.getSlideCount()
+    let i = 0
+    const slides = document.getElementsByClassName('remark-slide-container')
+    slides.forEach((slide) => {
+      if (i + 1 <= nslides) {
+        const canvasDiv = this.createCanvas(i)
+        slide.appendChild(canvasDiv)
+        i += 1
+      }
+    })
+
+    // Convert <canvas> to fabric.Canvas
+    const allCanvases = document.querySelectorAll('.xe-scribble.drawing-canvas')
+    this.fabrics = new Array()
+    allCanvases.forEach((el, index) => {
+      this.fabrics[index] = new fabric.Canvas(el.id, {
+        isDrawingMode: false,
+        containerClass: 'xe-scribble canvas-container',
+      })
+    })
+    this.currFabric = this.fabrics[0] // set to first slide to start
+  }
+
+  resizeContent() {
+    const scalerSize = document
+      .querySelector('.remark-visible .remark-slide-scaler')
+      .getBoundingClientRect()
+
+    // Resize canvas container
+    const outerContainers = document.querySelectorAll(
+      '.xe-scribble.outer-canvas-container',
+    )
+    outerContainers.forEach((div) => {
+      div.style.width = scalerSize.width + 'px'
+      div.style.height = scalerSize.height + 'px'
+      div.style.left = scalerSize.left + 'px'
+      div.style.top = scalerSize.top + 'px'
+    })
+
+    // Resize canvas itself
+    this.fabrics.forEach((fab) => {
+      fab.setWidth(scalerSize.width)
+      fab.setHeight(scalerSize.height)
+    })
+  }
+
+  fetchIcon(name) {
```

Comment:
Instead of a function with switch, you can add the svgs to the class in an object. After the class declaration, you would call

```js
Scribble.prototype.svgs = {
  draw: '<svg>...',
  eraser: '<svg>...',
  trash: '<svg>...'
}
```

and within the methods you would use `this.svgs['draw']` to get the icon svg.